src/v/security/oidc_authenticator.cc

```diff
@@ -134,4 +134,55 @@ authenticator::authenticate(std::string_view bearer_token) {
     return _impl->authenticate(bearer_token);
 }
 
+std::ostream& operator<<(std::ostream& os, sasl_authenticator::state const s) {
+    using state = sasl_authenticator::state;
+    switch (s) {
+    case state::init:
+        return os << "init";
+    case state::complete:
+        return os << "complete";
+    case state::failed:
+        return os << "failed";
+    }
+}
+
+sasl_authenticator::sasl_authenticator(service& service)
+  : _authenticator{service} {}
+
+sasl_authenticator::~sasl_authenticator() = default;
+
+ss::future<result<bytes>> sasl_authenticator::authenticate(bytes auth_bytes) {
+    if (_state != state::init) {
+        vlog(seclog.debug, "invalid oidc state: {}", _state);
```

Comment:
nit: I'd raise this to warn level

Reply:
I think in general all of these error messages should be INFO at lowest.

Reply:
> I think in general all of these error messages should be INFO at lowest.

I think I might just redo the logging a bit so that it can propagate with sufficient context.

Reply:
Logging improved

Reply:
Upped to warn